### ISSUE TITLE
Defining extern Rust items in separate file/crate (patterns for code reuse)

### ISSUE THREAD

**Issue by tychedelia** [dtolnay/cxx#1146]
I'm using CXX to develop plugins for an application I use, using the `Box<dyn Trait>` pattern described here https://github.com/dtolnay/cxx/pull/672. This works great, but I'm trying to figure out a way that I can library-ize the FFI definitions so they can be used in a bunch of different plugin sub-crates.  Essentially, I have a bridge defined like:

```rust
#[cxx::bridge]
mod ffi {
    // ... relevant other definitions used for our trait
    extern "Rust" {
        fn plugin_new() -> BoxDynPlugin;
    }
}

trait Plugin {}
```

and I'd like to be able to do something like to define the extern Rust item from another crate:
```rust
macro_rules! plugin {
    ($plugin_ty:ty) => {
        fn plugin_new() -> Box<dyn Plugin> {
            Box::new(<$plugin_ty>::new())
        }
    }
}
```

So that in my different plugin projects I could just import the macro into a sub-crate `lib.rs` and do:
```rust
struct MyPlugin;
impl Plugin for MyPlugin {}
plugin!(MyPlugin);
```

Basically, I want reuse the bridge to avoid having to copy it into each new plugin crate that I define. Are there any patterns for this kind of reuse?

Apologies if this has been asked before, wasn't able to dig anything up.

**Comment by tychedelia** [comment]
Missed this: https://github.com/dtolnay/cxx/issues/934. Doesn't seem like this is currently possible. Not sure what approach I'll use.
